amazon/ionbenchmark/benchmark_spec.py

```diff
@@ -177,12 +181,23 @@ def get_input_file_size(self):
 
     def get_data_object(self):
         """
-        Get the data object to be used for testing. Used for benchmarks that write data.
+        Get a generator that holds all data objects to be used for testing.
         """
         if not self._data_object:
             loader = self.get_loader_dumper()
-            with open(self.get_input_file(), "rb") as fp:
-                self._data_object = loader.load(fp)
+            format_option = self.get_format()
+            read_file = self.get_input_file()
+            if format_is_protobuf(format_option):
+                # Refer to https://github.com/amazon-ion/ion-python/issues/326
+                raise NotImplementedError("Benchmarking Protocol Buffer multiple top level object use case may not "
+                                          "support yet.")
+            else:
+                rtn = []
+                with open(read_file, 'br' if format_is_bytes(format_option) else 'r') as fp:
+                    obj = loader.load(fp)
+                    for v in obj:
+                        rtn.append(v)
```

Comment:
```suggestion
                    rtn = [v for v in loader.load(fp)]
```

Shorter, and I believe, faster.

Reply:
Changed, thanks for catching this up.